### DISCUSSION TITLE
Tax Rate Configuration

### DISCUSSION THREAD

**Discussion by pevey** [vendurehq/vendure#1948] [General]
Hello.

I have a store that is currently using Magento 2, and I've been exploring options for moving away from it.  

I'm in the US and in a state that requires collection of sales tax at the county level.  That's _county_, not country, just to be clear.  Each county might have multiple different tax zones. It's a mess.

In Magento 2, I can handle this by creating a different tax zones based on zip code, zip code range, or state-wide.  It looks like for now Vendure only handles country-wide rates.  That pretty much makes it a no go for any store based in the US.  Even if you are not in a state like mine that has arcane county-level collection requirements, you would at least need to be able to enter state-by-state rates.

Is there any workaround for this?  If not, is it by chance on the roadmap to provide more options around setting tax rates?

Thanks!

**Comment by michaelbromley** [comment]
Hi!

So the default setup of taxes is indeed at the country level in Vendure, which is suitable in the EU where we are based. But I am aware that the US has an ... interesting ... tax regime 😅. So for that reason we have the concept of the [TaxLineCalculationStrategy](https://www.vendure.io/docs/typescript-api/tax/tax-line-calculation-strategy/) - this is the logic that determines the sales tax applicable on a given order line. The default strategy indeed uses the built-in zone-based tax rates ([here's the super simple implementation](https://github.com/vendure-ecommerce/vendure/blob/master/packages/core/src/config/tax/default-tax-line-calculation-strategy.ts#L12)), but you can pretty much use any kind of logic you need in your own strategies. For instance, you could create a plugin which defines a custom entity to hold tax data by zip code area, and use that as a lookup in your own custom strategy.

Longer-term, there are some ideas around making the zone system more granular to support arbitrary definition of [Provinces](https://github.com/vendure-ecommerce/vendure/issues/76).

I hope that helps, and if you have any more questions, I invite you to join our [Slack group](https://join.slack.com/t/vendure-ecommerce/shared_invite/zt-1exzio25w-vjL5TYkyJZjK52d6jkOsIA) where you usually get pretty quick answers.

**Comment by RdAllen** [comment]
这是来自QQ邮箱的假期自动回复邮件。您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by gulangzai** [comment]
来件收悉！ 谢谢您。（望江县应急局）
